Library/Homebrew/test/test_language_python.rb

```diff
@@ -71,18 +73,24 @@ def test_pip_install_accepts_resource
     @venv.pip_install res
   end
 
-  def test_link_scripts_links_scripts
-    bin = (@dir/"bin")
-    dest = (@dir/"dest")
-    bin.mkpath
-    refute((bin/"kilroy").exist?)
-    refute((dest/"kilroy").exist?)
+  def test_pip_install_and_link_links_scripts
+    bin = (@dir/"bin").tap(&:mkpath)
+    dest = @formula.dest
+
+    refute_predicate bin/"kilroy", :exist?
+    refute_predicate dest/"kilroy", :exist?
+
     FileUtils.touch bin/"irrelevant"
-    @venv.link_scripts(dest) { FileUtils.touch bin/"kilroy" }
-    assert((bin/"kilroy").exist?)
-    assert((dest/"kilroy").exist?)
-    assert((dest/"kilroy").symlink?)
+    @formula.expects(:system).returns(true).with do |*params|
+      FileUtils.touch bin/"kilroy"
+      params.first == @dir/"bin/pip" && params.last == "foo"
+    end
+    @venv.pip_install_and_link "foo"
+
+    assert_predicate bin/"kilroy", :exist?
+    assert_predicate dest/"kilroy", :exist?
+    assert_predicate dest/"kilroy", :symlink?
     assert_equal((bin/"kilroy").realpath, (dest/"kilroy").realpath)
-    refute((dest/"irrelevant").exist?)
+    refute_predicate dest/"irrelevant", :exist?
```

Comment:
Because it results in a clearer message if the assertion fails. Instead of just saying “assertion failed” (no message is provided) it dumps the `Pathname` instance and the predicate of that instance that was supposed to be true/false.